### DISCUSSION TITLE
[Feature] Collaborative Albums

### DISCUSSION THREAD

**Discussion by xMaximillianous** [immich-app/immich#5649] [Feature Request]
### The feature

Hello!

I like the fact that Shared Albums do not allow others to add/edit photos, however It would be amazing to have a feature that would allow multiple people sharing an album have the ability to upload photos. This could be distinguished by calling it a Collaborative Album.

It would be really helpful in scenarios where multiple users (say going on a trip, etc.) are able to upload their share of the pictures taken to the same collaborative album without having to share multiple different albums for the same event from different users.

Thanks a lot for your hard work!

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by ghost** [comment]
I would love to see such a feature, I wanted to have something like this for my last family trip

**Comment by AngelaDMerkel** [comment]
It would be appreciated if this feature also allowed non-users to add a Pseudonym and upload their pictures to that. It's common when going on a trip with friends (for example) that they don't have their own account but nonetheless would still like to add their own images and participate

**Reply by alextran1502** [reply]
It can be done with the public shared link already, correct?

**Comment by brunokc** [comment]
I was about to submit a similar request when I found this. I had additional requirements though. Since this is so similar, I'm going to add them below. I'm also breaking it down into the individual pieces I thought would be needed to get to such a feature. Some of these may already be implemented:

- Support for sharing folders with upload permissions on a per user basis (authenticated via OIDC/OAuth2 against Microsoft, Google, etc). Sharing via link is not ideal here since it's easy to lose permission control if the link is forwarded.
   - Ideally, I shouldn't have to create a user in the system for this as some of these folks are temporary contributors. Having said that, some of them (particularly family members) may be already in the system.
- Support for identifying the uploader of each photo and allow operations (filtering, search, grouping, etc) by the uploader/contributor.
- By default, all contributors can also download any photo from the folder
- Optionally, allow setting a date range for the event (Dec 20 2025 through Jan 3 2026)
- Support in the mobile apps to auto-upload photos to the shared folder you're contributing to. This is in addition to any normal backup the mobile app already does. If a date range has been set, automatically stop uploading once the end date of the event is reached. 
- Allow user to either auto-upload all photos via mobile app or allow them to select which ones and let the app upload them in the background.
- Allow management of all this (add/remove users, add/remove photos, etc) from any endpoint (mobile app, web, etc).
